discovery/discovery.go

```diff
@@ -33,7 +39,7 @@ type WatchCallback func(entries []*Entry)
 
 // Discovery is exported
 type Discovery interface {
-	Initialize(string, uint64) error
+	Initialize(string, uint64, *TLS) error
```

Comment:
Since currently none of ou discovery service supports TLS, can we not change the prototype of this func but change it when we will add TLS ?